File: 1_Two_Sum/initialSolution.go
package main

func twoSum(nums []int, target int) []int {
	alreadySeen := make(map[int]int)
	resultIndex := make([]int, 2)

	for i := 0; i <= len(nums)-1; i++ {
		rest := target - nums[i]
		index, isPresent := alreadySeen[rest]
		if isPresent {
			resultIndex[0] = index
			resultIndex[1] = i
			return resultIndex
		} else {
			alreadySeen[nums[i]] = i
		}
	}
	return resultIndex
}

File: 1_Two_Sum/initialSolution.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> alreadySeen = new HashMap<>();
        int[] resultIndexes = new int[2];
        for (int i = 0; i <= nums.length - 1; i++) {
            int restThatsNeeded = target - nums[i];
            if (alreadySeen.containsKey(restThatsNeeded)) {
                resultIndexes[0] = alreadySeen.get(restThatsNeeded);
                resultIndexes[1] = i;
                return resultIndexes;
            } else {
                alreadySeen.put(nums[i], i);
            }

        }
        return resultIndexes;
    }
}
